src/YDialogSpy.cc

```diff
@@ -278,70 +474,427 @@ void fillTree( YWidgetTreeItem * 		parent,
+                cb->addItems( add_items );
+            }
+        }
+        else if (type == "Empty")
+            editWidget(f->createEmpty(widget));
+        else if (type == "Frame")
+            editWidget(f->createFrame(widget, "Frame"));
+        else if (type == "HBox")
+            editWidget(f->createHBox(widget));
+        else if (type == "Heading")
+            editWidget(f->createHeading(widget, "Heading"));
+        else if (type == "HSpacing")
+            editWidget(f->createHSpacing(widget));
+        else if (type == "HStretch")
+            editWidget(f->createHStretch(widget));
+        else if (type == "CheckBox")
+            editWidget(f->createCheckBox(widget, "Check Box"));
+        else if (type == "CheckBoxFrame")
+            // make it checked by default
+            editWidget(f->createCheckBoxFrame(widget, "Check Box Frame", true));
+        else if (type == "Image")
+            editWidget(f->createImage(widget, ""));
+        else if (type == "InputField")
+            editWidget(f->createInputField(widget, "Input"));
+        else if (type == "IntField")
+            editWidget(f->createIntField(widget, "Integer Field", 0, 100, 50));
+        else if (type == "Label")
+            editWidget(f->createLabel(widget, "Label"));
+        else if (type == "Left")
+            editWidget(f->createLeft(widget));
+        else if (type == "LogView")
+            editWidget(f->createLogView(widget, "Log View", 12));
+        else if (type == "MenuButton")
+        {
+            auto menu = f->createMenuButton( widget, "Menu" );
+            editWidget(menu);
+
+            YPopupInternal::StringArray arr { "", "", "" };
+
+            // edit the item list and update the widget after pressing OK
+            if (YPopupInternal::editStringArray(arr, "Menu Items"))
+            {
+                YItemCollection add_items;
+                // access by reference
+                for(auto&& str: arr) add_items.push_back( new YMenuItem( str ) );
+                menu->addItems( add_items );
+            }
+        }
+        else if (type == "MinHeight")
+            editWidget(f->createMinHeight(widget, 10));
+        else if (type == "MinWidth")
+            editWidget(f->createMinWidth(widget, 10));
+        else if (type == "MinSize")
+            editWidget(f->createMinSize(widget, 10, 10));
+        else if (type == "MultiLineEdit")
+            editWidget(f->createMultiLineEdit(widget, "MultiLineEdit"));
+        else if (type == "MultiSelectionBox")
+        {
+            auto msb = f->createMultiSelectionBox(widget, "MultiSelection Box");
+            editWidget(msb);
+
+            YPopupInternal::StringArray arr { "", "", "" };
+
+            // edit the item list and update the widget after pressing OK
+            if (YPopupInternal::editStringArray(arr, "Items"))
+                // access by reference
+                for(auto&& str: arr) msb->addItem(str);
```

Comment:
These code blocks could be simplified if we had a `editNewStringArray(title)` method that would declare the array and move it out to the caller. If the user canceled it would return an empty one.
Then we could have
```cpp
auto msb = f->createMultiSelectionBox(widget, "MultiSelection Box");
editWidget(msb);
for(auto&& str: editNewStringArray("Items"))
  msb->addItem(str);
```